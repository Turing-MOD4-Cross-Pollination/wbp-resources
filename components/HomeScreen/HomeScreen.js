/* eslint-disable react/jsx-filename-extension */
import React, { Component } from 'react';
import { StyleSheet, Text, View, TouchableOpacity, Image, Dimensions } from 'react-native';

const { height, width } = Dimensions.get('window');

export default class HomeScreen extends Component {
  static navigationOptions = {
    title: 'Home',
    headerStyle: {
      backgroundColor: '#4E5A31',
      elevation: 0,
      shadowOpacity: 0,
      borderBottomWidth: 0,
    },
    headerTintColor: '#fff',
    headerTitleStyle: {
      fontWeight: 'bold',
    },
  };

  render = () => {
    const { navigate } = this.props.navigation;
    return (
      <>
        <View style={styles.container}>
          <View style={styles.logoContainer}>
            <Image source={require('../../assets/header_logo.png')} style={styles.logoImage} />
          </View>
          {/* <Text style={styles.colorado}>Women's Bean Project</Text> */}
          <Text style={styles.summary}>Changing women's lives.</Text>
          <View style={styles.line} />
          <View style={styles.buttonsContainer}>
            <View style={styles.row}>
              <TouchableOpacity style={{ width: '48%' }} onPress={() => navigate('Now')}>
                <View style={styles.buttonNow} title="This is a button">
                  <Text style={{ ...styles.buttonText, color: 'white' }}>Now</Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity style={{ width: '48%' }} onPress={() => navigate('NearMe')}>
                <View style={styles.buttonNearMe} title="This is a button">
                  <Text style={styles.buttonText}>Near Me</Text>
                </View>
              </TouchableOpacity>
            </View>
            <TouchableOpacity style={{ width: '100%' }} onPress={() => navigate('ForMe')}>
              <View style={styles.buttonForMe} title="This is a button">
                <Text style={{ ...styles.buttonText, fontSize: 35 }}>My Resources</Text>
              </View>
            </TouchableOpacity>
          </View>
        </View>
      </>
    );
  };
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#4E5A31',
    alignItems: 'center',
    justifyContent: 'space-around',
    height: '100%',
    flexDirection: 'column',
    paddingTop: 40,
    margin: 0,
    width,
  },
  buttonsContainer: {
    flexDirection: 'column',
    alignItems: 'center',
    height: 173,
    justifyContent: 'space-between',
    width: '83%',
    marginBottom: 100,
  },
  buttonNow: {
    height: 80,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    backgroundColor: '#B14F1F',
    borderRadius: 12,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.3,
    shadowRadius: 4.65,
    elevation: 7,
  },
  buttonNearMe: {
    height: 80,
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#FFF',
    borderRadius: 12,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.3,
    shadowRadius: 4.65,
    elevation: 7,
    padding: 5,
  },
  buttonForMe: {
    height: 80,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    backgroundColor: '#FFF',
    borderRadius: 12,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.3,
    shadowRadius: 4.65,

    elevation: 7,
  },
  row: {
    flexDirection: 'row',
    width: '100%',
    justifyContent: 'space-between',
    alignItems: 'center',
    color: 'white',
  },
  buttonText: {
    fontSize: 32,
    fontWeight: '700',
    textAlign: 'center',
    color: '#4E5A31',
  },
  line: {
    width: '83%',
    backgroundColor: '#fff',
    height: 1.5,
    marginBottom: 20,
  },
  logoContainer: {
    backgroundColor: 'white',
    width: '100%',
    height: 150,
    marginBottom: 50,
    display: 'flex',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  logoText: {
    fontSize: 40,
    width: 230,
    marginLeft: 160,
    fontWeight: '700',
    height: 50,
    color: '#003180',
    letterSpacing: -1,
  },
  logoImage: {
    width: '90%',
    resizeMode: 'contain',
  },
  colorado: {
    fontSize: 28,
    color: 'white',
    fontWeight: 'bold',
  },
  summary: {
    color: 'white',
    width: '83%',
    fontSize: 20,
    marginBottom: 15,
    textAlign: 'center',
    opacity: 0.8,
  },
});
