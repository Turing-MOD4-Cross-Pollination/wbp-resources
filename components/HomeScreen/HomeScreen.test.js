import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import HomeScreen from './HomeScreen';

describe('HomeScreen', () => {
  let wrapper;
  const navigation = { navigate: jest.fn() };

  beforeEach(() => {
    wrapper = shallow(<HomeScreen navigation={navigation} />);
  });

  it('App should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('HomeScreen', () => {
  it('renders correctly', () => {
    const navigation = jest.fn();

    const tree = renderer.create(<HomeScreen navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
