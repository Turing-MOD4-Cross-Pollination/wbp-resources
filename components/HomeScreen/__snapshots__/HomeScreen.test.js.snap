// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen App should match snapshot 1`] = `ShallowWrapper {}`;

exports[`HomeScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#003180",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "space-around",
      "margin": 0,
      "paddingTop": 40,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "display": "flex",
        "height": 70,
        "justifyContent": "space-around",
        "marginBottom": 50,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/logo.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 140,
          "position": "absolute",
          "right": "61%",
          "top": "-54%",
          "width": 140,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#003180",
          "fontSize": 40,
          "fontWeight": "700",
          "height": 50,
          "letterSpacing": -1,
          "marginLeft": 160,
          "width": 230,
        }
      }
    >
      community
    </Text>
    <Text
      style={
        Object {
          "color": "#003180",
          "fontSize": 40,
          "fontWeight": "700",
          "height": 50,
          "letterSpacing": -1,
          "marginLeft": 160,
          "marginTop": -15,
          "width": 230,
        }
      }
    >
      connect
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 28,
        "fontWeight": "bold",
      }
    }
  >
    COLORADO
  </Text>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "marginBottom": 15,
        "opacity": 0.8,
        "textAlign": "center",
        "width": "83%",
      }
    }
  >
    Find all resources available and make them your own for fast and easy access.
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "height": 1.5,
        "marginBottom": 20,
        "width": "83%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 173,
        "justifyContent": "space-between",
        "marginBottom": 100,
        "width": "83%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "color": "white",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "48%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ed3b53",
              "borderRadius": 12,
              "display": "flex",
              "elevation": 7,
              "height": 80,
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 4.65,
              "width": "100%",
            }
          }
          title="This is a button"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 32,
                "fontWeight": "700",
                "textAlign": "center",
              }
            }
          >
            Now
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "48%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFF",
              "borderRadius": 12,
              "display": "flex",
              "elevation": 7,
              "height": 80,
              "justifyContent": "center",
              "padding": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 4.65,
              "width": "100%",
            }
          }
          title="This is a button"
        >
          <Text
            style={
              Object {
                "color": "#003180",
                "fontSize": 32,
                "fontWeight": "700",
                "textAlign": "center",
              }
            }
          >
            Near Me
          </Text>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "borderRadius": 12,
            "display": "flex",
            "elevation": 7,
            "height": 80,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4.65,
            "width": "100%",
          }
        }
        title="This is a button"
      >
        <Text
          style={
            Object {
              "color": "#003180",
              "fontSize": 35,
              "fontWeight": "700",
              "textAlign": "center",
            }
          }
        >
          My Resources
        </Text>
      </View>
    </View>
  </View>
</View>
`;
